x-pack/plugins/ml/public/application/trained_models/models_management/delete_models_modal.tsx

```diff
@@ -18,18 +19,58 @@ import {
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { ModelItemFull } from './models_list';
+import { useTrainedModelsApiService } from '../../services/ml_api_service/trained_models';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+import { DeleteSpaceAwareItemCheckModal } from '../../components/delete_space_aware_item_check_modal';
 
 interface DeleteModelsModalProps {
   models: ModelItemFull[];
-  onClose: (deletionApproved?: boolean) => void;
+  onClose: (refreshList?: boolean) => void;
 }
 
 export const DeleteModelsModal: FC<DeleteModelsModalProps> = ({ models, onClose }) => {
-  const modelsWithPipelines = models
-    .filter((model) => !!model.pipelines)
-    .map((model) => model.model_id);
+  const trainedModelsApiService = useTrainedModelsApiService();
+  const { displayErrorToast, displaySuccessToast } = useToastNotificationService();
 
-  return (
+  const [canDeleteModel, setCanDeleteModel] = useState(false);
+
+  const modelsWithPipelines = useMemo(
```

Comment:
I am seeing this error when trying to delete a model from a DFA job which is only tagged in a single space. I'm guessing the check around pipelines is wrong as this model does not have a pipeline. Also the error (when relevant) needs cleaning up.

![image](https://user-images.githubusercontent.com/7405507/158170182-87c5a4d0-9aba-4b04-8d59-c6ee1374c204.png)

Reply:
This issue exists on main too.
I'm not sure what the expected behaviour is supposed to be.
If a model does have associated pipelines, it is blocked from being deleted through the UI.
<img width="237" alt="image" src="https://user-images.githubusercontent.com/22172091/158214366-5fe763b0-98e4-44f0-a1d1-6b4ac8b4f3ea.png">


I've fixed this incorrectly appearing callout, but it looks like the callout should be removed entirely as it'll never be seen.

Perhaps @darnautov has some idea of the correct behaviour?

Reply:
Originally it was possible to select multiple models and perform a bulk-delete action. So in case some of the selected models had associated pipelines, this warning callout was shown.

Reply:
Is it still needed now?
It's not possible to select models with pipelines.

Reply:
ok, then it's not needed anymore

Reply:
https://github.com/elastic/kibana/pull/127438/commits/93b03665269dd795ec9ca1eb2f6a27ee266f4ff5